packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1079,6 +1079,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         this.summaryCollection = new SummaryCollection(this.deltaManager, this.logger);
 
+        const { attachState, pendingLocalState } = this.context;
+        this.dirtyContainer = attachState !== AttachState.Attached
+            || (pendingLocalState as IPendingLocalState)?.pendingStates.length > 0;
```

Comment:
The calls back and forth between the context/runtime on pending local state are confusing to me, e.g. I don't understand why we wouldn't directly call `this.getPendingLocalState()`.  Maybe this area (pending state) needs some more attention in general in the future?

Reply:
Good points @ChumpChief. Interestingly, `PendingStateManager` does not consider this initial state (`pendingLocalState` coming from context) as pending ops. Only at a later time, when stashed ops are applied at their reference sequence is when initial state transitions to local state within `PendingStateManager`, and is actually considered pending. So, requesting pending ops count on the initialized `PendingStateManager` in the runtime constructor would just give back 0, even though `stashed` ops is present. I was thinking well let's expose this added API on pending manager (like getStashedOps or getShashedOpsCount), but then we introduce another term and another API with somewhat limited purpose.

Reply:
@wes-carlson .... if you have any thoughts on the above.

Reply:
@wes-carlson if you think Pending Manager is work in progress (Im not sure what further complexities you are thinking about), do you think making related pending manager changes is appropriate to do it as part of this PR (if so, happy to do it here). Or keep things as they are in this PR and rely on context as done above ?

Reply:
I'll leave the pending manager change out of this PR then, as it needs a bit more work there unrelated to dirty flag. From the perspective of runtime constructor, looking at `pendingLocalState` is enough to (correctly) determine there are pending ops present.

Reply:
https://github.com/microsoft/FluidFramework/issues/8893

Reply:
Can we create an issue to track the work in pending state manager?